### PR TITLE
drivers: modem: hl7800: fix file open in fw update

### PR DESCRIPTION
**Pull request by rerickson1**
File open function for firmware update requires a flags parameter to be passed in.

### CODE REVIEW

**Comment by rerickson1** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/28745

**Review by galak — CHANGES_REQUESTED** [review]
Fix is fine, but we should add a build test to `tests/drivers/build_all`

**Comment by rerickson1** [comment]
> Fix is fine, but we should add a build test to `tests/drivers/build_all`

Sounds good. What are your thoughts on how to approach this? 
1. Add a board file overlay in the boards folder? (pinnacle_100_dvk can be used to test the HL7800 driver).
2. Add a new hl7800.conf that exercises all HL7800 driver options? testcase updated to use hl7800.conf with board pinnacle_100_dvk.

**Comment by galak** [comment]
> Sounds good. What are your thoughts on how to approach this?
> 
> 1. Add a board file overlay in the boards folder? (pinnacle_100_dvk can be used to test the HL7800 driver).
> 2. Add a new hl7800.conf that exercises all HL7800 driver options? testcase updated to use hl7800.conf with board pinnacle_100_dvk.

I don't know if we can get away with doing something similar to `i2c.dtsi` called `modem.dtsi` and have the hl7800, ublox-sara-r4, and wncm14a2a in there.  Than have a `modem.conf` that enables those drivers (similar to `ethernet.conf`).

**Comment by mike-scott** [comment]
Change looks good, and I agree with @galak we should have a build test of this driver.

**Comment by rerickson1** [comment]
> > Sounds good. What are your thoughts on how to approach this?
> > 
> > 1. Add a board file overlay in the boards folder? (pinnacle_100_dvk can be used to test the HL7800 driver).
> > 2. Add a new hl7800.conf that exercises all HL7800 driver options? testcase updated to use hl7800.conf with board pinnacle_100_dvk.
> 
> I don't know if we can get away with doing something similar to `i2c.dtsi` called `modem.dtsi` and have the hl7800, ublox-sara-r4, and wncm14a2a in there. Than have a `modem.conf` that enables those drivers (similar to `ethernet.conf`).

Due to the driver relying on a UART and multiple IO, couldn't see how to do it like that. Added the test for the HL7800 driver and made use of the pinnacle_100_dvk because it already supports the HL7800.

**Comment by rerickson1** [comment]
> Change looks good, and I agree with @galak we should have a build test of this driver.

@mike-scott get I get a re-review, thanks!

**Review by jukkar — APPROVED** [review]
LGTM

**Review by mike-scott — APPROVED** [review]

**Comment by rerickson1** [comment]
@galak can I get a second look and get this merged?

**Review by galak — APPROVED** [review]

**Review by galak — CHANGES_REQUESTED** [review]
can you rename `modem_hl7800.conf` to modem.conf

File: tests/drivers/build_all/testcase.yaml [hunk 1]
```diff
@@ -84,3 +84,8 @@ tests:
     extra_args: CONF_FILE=dac.conf
     min_ram: 32
     depends_on: dac spi
+  drivers.modem.hl7800.build:
```
Comment (galak) [inline tests/drivers/build_all/testcase.yaml:87]: rename to `drivers.modem.build`

**Comment by rerickson1** [comment]
> can you rename `modem_hl7800.conf` to modem.conf

are you sure? the settings in there are specific to the hl7800 modem. Different modems may have different settings.

**Comment by galak** [comment]
> are you sure? the settings in there are specific to the hl7800 modem. Different modems may have different settings.

I am, as I can follow up this PR w/having the test build support for multiple modems at the same time.

File: tests/drivers/build_all/testcase.yaml (hunk 1 above)
Comment (mike-scott) [reply tests/drivers/build_all/testcase.yaml:87]: > rename to `drivers.modem.build`

What if I want to add tests for another modem?

Comment (galak) [reply tests/drivers/build_all/testcase.yaml:87]: than we'll do that like we add for different sensors.  Is there something that is mutually exclusive about the modem drivers?

**Comment by galak** [comment]
@rerickson1 / @mike-scott here's a PR based on the build_all change in here that enables additional modem support:

https://github.com/zephyrproject-rtos/zephyr/pull/29211

I would like to squash it down to one commit for the build_all changes, and one for the build bug fix, but left it as is for now for review purposes (and signed-off-by ack if I squash it all down) from @rerickson1 .

**Review by galak — APPROVED** [review]

**Pull request merged**